### PR TITLE
XRT: linux-driver: Fix mmap on MIPS and build error

### PR DESCRIPTION
**Pull request by AlooGS**

### CODE REVIEW

File: src/runtime_src/core/pcie/driver/linux/xocl/lib/libqdma/qdma_debugfs.h [hunk 1]
```diff
@@ -23,7 +23,11 @@
 #include <linux/pci.h>
 #include <linux/debugfs.h>
 #include <linux/fs.h>
+#if defined(__mips__)
+#include <linux/uaccess.h>
+#else
 #include <asm/uaccess.h>
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/driver/linux/xocl/lib/libqdma/qdma_debugfs.h:29]: I think we should just include linux/uaccess.h in all cases. @karenx-xilinx , do you see any problem with including linux/uaccess.h here?

**Comment by maxzhen** [comment]
ok to test

File: src/runtime_src/core/pcie/driver/linux/xocl/lib/libqdma/qdma_debugfs.h (hunk 1 above)
Comment (karenx-xilinx) [reply src/runtime_src/core/pcie/driver/linux/xocl/lib/libqdma/qdma_debugfs.h:29]: Yes, agree. we can just include uaccess.h always.

Comment (AlooGS) [reply src/runtime_src/core/pcie/driver/linux/xocl/lib/libqdma/qdma_debugfs.h:29]: I do not why use asm/uaccess.h, i also think should use linux/uaccess.h in all arches.

**Comment by AlooGS** [comment]
> where is "SZ_4G" defined?

It defined in linux/sizes.h, my kernel base code version is 4.19. Can you tell me your kernel base ?

**Comment by maxzhen** [comment]
> 
> 
> > where is "SZ_4G" defined?
> 
> It defined in linux/sizes.h, my kernel base code version is 4.19. Can you tell me your kernel base ?

it does not seem to be available on centos 7.4 with kernel version 3.10.0. you may need to define a private one for it to work with other kernel versions.

**Comment by AlooGS** [comment]
> where is "SZ_4G" defined?
> 
> It defined in linux/sizes.h, my kernel base code version is 4.19. Can you tell me your kernel base ?
> 
> it does not seem to be available on centos 7.4 with kernel version 3.10.0. you may need to define a private one for it to work with other kernel versio

**Comment by AlooGS** [comment]
> where is "SZ_4G" defined?
> It defined in linux/sizes.h, my kernel base code version is 4.19. Can you tell me your kernel base ?
> it does not seem to be available on centos 7.4 with kernel version 3.10.0. you may need to define a private one for it to work with other kernel versio

Can you help confirm it in linux/sizes.h in your kernel ?

**Comment by maxzhen** [comment]
> 
> 
> > where is "SZ_4G" defined?
> > It defined in linux/sizes.h, my kernel base code version is 4.19. Can you tell me your kernel base ?
> > it does not seem to be available on centos 7.4 with kernel version 3.10.0. you may need to define a private one for it to work with other kernel versio
> 
> Can you help confirm it in linux/sizes.h in your kernel ?

the build is failing on pipeline test for your pull request due to missing SZ_4G macro. it seems that this macro is introduced in 4.17 kernel. so it will not be available for 3.10.

**Comment by AlooGS** [comment]
> where is "SZ_4G" defined?
> It defined in linux/sizes.h, my kernel base code version is 4.19. Can you tell me your kernel base ?
> it does not seem to be available on centos 7.4 with kernel version 3.10.0. you may need to define a private one for it to work with other kernel versio
> 
> Can you help confirm it in linux/sizes.h in your kernel ?
> 
> the build is failing on pipeline test for your pull request due to missing SZ_4G macro. it seems that this macro is introduced in 4.17 kernel. so it will not be available for 3.10.

Thank you very much, i have fix it.

**Pull request closed without merging**